src/inject.js

```diff
@@ -44,12 +44,21 @@ const injectStyleTag = (cssContents /* : string */) => {
         }
     }
 
+    const sheet = ((styleTag.styleSheet || styleTag.sheet /* : any */) /* : CSSStyleSheet */);
 
-    if (styleTag.styleSheet) {
-        // $FlowFixMe: legacy Internet Explorer compatibility
-        styleTag.styleSheet.cssText += cssContents;
+    if (sheet.insertRule) {
+        cssContents.forEach((rule) => {
+            try {
+                sheet.insertRule(rule, sheet.cssRules.length);
```

Comment:
It might be worth moving this out into a variable that we increment, to avoid having to read this value every time.

```js
let numberOfRules = sheet.cssRules.length;
cssContents.forEach((rule) => {
  try {
    sheet.insertRule(rule, numberOfRules);
    numberOfRules += 1;
  } catch(e) { ... }
});
```